### PR TITLE
[test] Publish MAGMA tests and adapt to PE 18.08

### LINKED ISSUE #186: Remove custom bits from MAGMA tests

### LINKED ISSUE #364: Adapt MAGMA tests to PE 18.07

### LINKED ISSUE #389: Adapt MAGMA tests to the new build systems feature

### PR DESCRIPTION
**Pull request by ajocksch**
Fixes #364.
Fixes #389.
Fixes #186.

### CODE REVIEW

**Comment by ajocksch** [comment]
magma 2.4 needs to be installed first

File: cscs-checks/libraries/magma/magma_checks.py [hunk 1]
```diff
@@ -1,279 +1,197 @@
...
+            self.perf_patterns = {
+                'magma': sn.extractsingle(r'MAGMA GFlops: (?P<magma_gflops>\S+)',
+                                          self.stdout, 'magma_gflops', float),
+                'cublas': sn.extractsingle(
+                    r'cuBLAS GFlops: (?P<cublas_gflops>\S+)', self.stdout,
+                    'cublas_gflops', float),
+                'cpu': sn.extractsingle(r'CPU GFlops: (?P<cpu_gflops>\S+)',
+                                        self.stdout, 'cpu_gflops', float)
+            }
+            if variant== 'prod':
```
Comment (vkarak) [inline cscs-checks/libraries/magma/magma_checks.py]: Please fix formatting here.

File: cscs-checks/libraries/magma/magma_checks.py [hunk 2]
```diff
@@ -1,279 +1,197 @@
...
         self.prebuild_cmd = ['patch < patch.txt']
-        self.modules = ['cudatoolkit', 'magma']
         self.build_system = 'Make'
-        self.build_system.makefile = makefile
+        self.build_system.makefile = 'Makefile_%s' % name
         # Compile with -O0 since with a higher level the compiler seems to
         # optimise something away
         self.build_system.cflags = ['-O0']
-        self.build_system.cxxflags = ['-O0']
+        self.build_system.cxxflags = ['-O0', '-std=c++11']
```
[the remark this answers is not in the document]
Comment (ajocksch) [reply cscs-checks/libraries/magma/magma_checks.py]: It has no affect on the library itself. It is written in the comment that higher optimisation levels do not work properly.

[the remark this answers is not in the document]
Comment (vkarak) [reply cscs-checks/libraries/magma/magma_checks.py]: Ok.

File: cscs-checks/libraries/magma/magma_checks.py [hunk 3]
```diff
@@ -1,279 +1,197 @@
...
+            self.sourcesdir = 'magma4'
+
+        if variant == 'prod':
+            self.tags |= {'production'}
+        elif variant == 'maint':
+            self.tags |= {'maintenance'}
+
+        if name == 'cblas_z':
+            self.perf_patterns = {
+                'value_duration':
```
[the remark this answers is not in the document]
Comment (ajocksch) [reply cscs-checks/libraries/magma/magma_checks.py]: done

File: cscs-checks/libraries/magma/magma_checks.py (hunk 1 above)
Comment (ajocksch) [reply cscs-checks/libraries/magma/magma_checks.py]: done

**Comment by vkarak** [comment]
This is PR is ready to merge. Simply waiting for https://github.com/eth-cscs/production/pull/827

**Pull request merged**